homeassistant/components/amcrest/camera.py

```diff
@@ -92,12 +106,129 @@ def __init__(self, hass, amcrest):
         finally:
             await stream.close()
 
+    # Entity property overrides
+
     @property
     def name(self):
         """Return the name of this camera."""
         return self._name
 
+    @property
+    def device_state_attributes(self):
+        """Return the Amcrest-spectific camera state attributes."""
+        return self._static_attrs
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORT_ON_OFF
+
+    # Camera property overrides
+
+    @property
+    def is_recording(self):
+        """Return true if the device is recording."""
+        return self._is_recording
+
+    @property
+    def brand(self):
+        """Return the camera brand."""
+        return 'Amcrest'
+
+    @property
+    def model(self):
+        """Return the camera model."""
+        return self._model
+
     @property
     def stream_source(self):
         """Return the source of the stream."""
         return self._camera.rtsp_url(typeno=self._resolution)
+
+    @property
+    def is_on(self):
+        """Return true if on."""
+        return self.video_enabled
+
+    # Additional Amcrest Camera properties
+
+    @property
+    def video_enabled(self):
```

Comment:
Eliminate this property.  It's only used in `is_on` where you can simply return `self._camera.video_enabled`

Reply:
See  comments on the corresponding setter. Also, if this property were to be removed, shouldn't is_on return self.is_streaming rather than interacting with the device?

Reply:
It is perfectly fine for it to return a value from the device.  It will only be called when updating state.

Reply:
`self._camera.video_enabled` doesn't retain any state from the device, but rather will communicate with the device to retrieve the information. It seems from your other feedback that this shouldn't be happening inside the `is_on` property. So I think the correct thing to do is to have `is_on` return the `self.is_streaming` attribute. Does that sound right?

Reply:
Correct then access of `self._camera.video_enabled` should be treated like a method and called from `update`.  If `is_streaming` means the same as `is_on` then yes that's fine to return.